src/foreign.c

```diff
@@ -2091,7 +2091,7 @@ int igraph_write_graph_pajek(const igraph_t *graph, FILE *outstream) {
                               };
     const char *estrnames2[] = { "a", "p", "l", "lc", "c" };
 
-    const char *newline = "\x0d\x0a";
```

Comment:
The goal with the original code was to ensure that we always print Windows-style line endings, even when we run the code on Linux or macOS. So I think it is correct here -- but it works only if the input file is opened in binary mode, not when it is opened in text mode. So, we need to fix the places where `igraph_write_graph_pajek()` is called in the test cases to ensure that the files are opened in binary mode, and also document this requirement in `igraph_write_graph_pajek()`.

Reply:
330f55511 addresses the problem in the Pajek tests cases where we now force `stdout` to be opened in binary mode.

Reply:
Unfortunately, using `#include <unistd.h>` in 330f555 is not portable.  

For the issue itself: Just out of curiosity - why should `igraph_write_graph_pajek()` write Windows-style EOLs on any platform? This appears to be an artificial requirement.  Is this because of the pajek lexer (not having looked into the dependencies here)?

Reply:
Earlier versions of Pajek insisted on CRLF line endings as it was (is) primarily a Windows-only tool. (You can use it on Linux or macOS with Wine, though). We've contacted the developers on the Pajek mailing list and it looks like recent versions are okay with either Unix or Windows style line endings, so we might just drop support for older Pajek versions and just use the platform-native style.

Reply:
That sounds great to me.

Reply:
330f555 is now reverted; I have simply switched to Unix line endings in 4d52c2e3d. This should produce:

* LF on Windows with binary mode
* CRLF on Windows with text mode
* LF on Unix, both in binary and in text mode